lib/web/websocket/websocket.js

```diff
@@ -516,27 +512,33 @@ class WebSocket extends EventTarget {
   }
 
   #onFail (code, reason) {
+    if (reason) {
+      // TODO: process.nextTick
+      fireEvent('error', this, (type, init) => new ErrorEvent(type, init), {
+        error: new Error(reason),
+        message: reason
+      })
+    }
+
     // If _The WebSocket Connection is Established_ prior to the point where
     // the endpoint is required to _Fail the WebSocket Connection_, the
     // endpoint SHOULD send a Close frame with an appropriate status code
     // (Section 7.4) before proceeding to _Close the WebSocket Connection_.
     if (isEstablished(this.#handler.readyState)) {
       this.#onClose(code, reason, reason?.length)
+    } else {
+      // If the WebSocket connection could not be established, it is also said
+      // that _The WebSocket Connection is Closed_, but not _cleanly_.
```

Comment:
this was the original bug - we were handling cases when the connection was established and then closed, but not cases when the connection was not established